plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -223,10 +208,14 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
                   result.prev_block = block_position{current_request->start_block_num - 1, *prev_block_id};
                if (current_request->fetch_block)
                   plugin->get_block(current_request->start_block_num, result.block);
-               if (current_request->fetch_traces && plugin->trace_log)
-                  plugin->get_log_entry(*plugin->trace_log, current_request->start_block_num, result.traces);
-               if (current_request->fetch_deltas && plugin->chain_state_log)
-                  plugin->get_log_entry(*plugin->chain_state_log, current_request->start_block_num, result.deltas);
+               if (current_request->fetch_traces && plugin->trace_log) {
+                  result.traces = plugin->trace_log->get_log_entry(current_request->start_block_num);
+               }
+               if (current_request->fetch_deltas && plugin->chain_state_log) {
+                  result.deltas = plugin->chain_state_log->get_entry(
+                      current_request->start_block_num,
+                      [](const bytes& data, uint32_t) { return state_history::zlib_decompress(data); });
+               }
```

Comment:
I like where this is headed in terms of moving all business logic into `state_history`. This exposes that it is zlib_decompressed. I would like a wrapper method for that in `state_history`.